packages/controllers/dataset_form.coffee

```diff
@@ -0,0 +1,33 @@
+if Meteor.isClient
+  Template.datasetForm.events
+    'submit form': (event, template) ->
+      event.preventDefault()
+      inputFile = event.target.file?.files[0]
+      unless inputFile
+        toastr.error("Error")
+      else
+        file = new FS.File inputFile
```

Comment:
Yeah, I initially tried something like that and ran into problems because the CollectionFS insert has to be done on the client for it to handle the upload. I'll give it another shot.